pkg/cmd/server/origin/run_components.go

```diff
@@ -376,7 +376,7 @@ func (c *MasterConfig) RunDeploymentImageChangeTriggerController() {
 // RunSDNController runs openshift-sdn if the said network plugin is provided
 func (c *MasterConfig) RunSDNController() {
 	oClient, kClient := c.SDNControllerClients()
-	controller, err := sdnfactory.NewMasterPlugin(c.Options.NetworkConfig.NetworkPluginName, oClient, kClient)
+	controller, err := sdnfactory.NewMasterPlugin(c.Options.NetworkConfig.NetworkPluginName, oClient, kClient, c.EtcdHelper)
```

Comment:
For a), I think I'd really recommend fitting into the new migrator framework and have the admins perform a manual step once the cluster is updated.

1. Have the registry read both sets of values (objects and namespaces) for Lists and Gets
2. Updates apply first to the namespace, then to the object (if one fails, fail the call, but don't revert the namespace), same for Deletes and Creates
2. If the objects are specified they take precedence over the namespace value
3. Clients continue unchanged (continue to use old client calls)
4. Migrator (part of the migrate command) copies the value over from the object to the namespace, and once it succeeds, deletes the object

I believe that will be backwards compatible and rollback compatible.